src/main/java/com/lukian/bookstore/service/cart/ShoppingCartServiceImpl.java

```diff
@@ -93,8 +106,15 @@ private CartItem getItemFromDb(Long cartItemId) {
     }
 
     private ShoppingCart getCartFromDB(Long userId) {
-        return cartRepository.findByUserId(userId)
-                .orElseThrow(() -> new EntityNotFoundException(
-                        "Cannot find cart by userId: " + userId));
+        Optional<ShoppingCart> cartFromDb = cartRepository.findByUserId(userId);
+        if (cartFromDb.isEmpty()) {
+            ShoppingCart shoppingCart = new ShoppingCart();
+            User userFromDb = userRepository.findById(userId)
+                    .orElseThrow(() -> new EntityNotFoundException(
+                            "User not found by user id: " + userId));
+            shoppingCart.setUser(userFromDb);
+            return cartRepository.save(shoppingCart);
+        }
+        return cartFromDb.get();
```

Comment:
you can simplify it. Find the users first and then use orElseGet() to get or create shopping cart